Data/Vector.hs

```diff
@@ -1776,8 +1776,19 @@ fromListN = G.fromListN
 
 -- | /O(1)/ Convert an array to a vector.
 fromArray :: Array a -> Vector a
+{-# INLINE fromArray #-}
```

Comment:
Although I originally wrote this function, I now feel that accepting offset and length would be better here. This gives the user a little more power. Of course, the drawback is that the user can segfault the program if they give bad boundaries.

Reply:
Same thing here, constructor is available for such use cases. It is better to follow it with a slice function, i.e.:
* safe: `slice offset size . fromArray`
* unsafe: `unsafeSlice offset size . fromArray`

Reply:
I agree with @lehins here. Functions that could violate memory safety should be clearly marked as such.